3.1.0 changelog (#2993)
AzureAD/microsoft-identity-web: changelog.md

3.1.0
=========
- Updated to Microsoft.IdentityModel.* 8.0.2

### Security improvement:

> - Id Web now uses `CaseSensitiveClaimsIdentity` by default and provides AppContextSwitches to fallback to using `ClaimsIdentity`. See PR [#2977](https://github.com/AzureAD/microsoft-identity-web/pull/2977) for details.

Reviewer: ```suggestion
- Id Web now uses `CaseSensitiveClaimsIdentity` by default and provides AppContextSwitches to fallback to using `ClaimsIdentity`. This means that when you loopup claims with FindFirst(), FindAll() and HasClaim(), you need to provide the right casing for the claim. See PR [#2977](https://github.com/AzureAD/microsoft-identity-web/pull/2977) for details.
```

### Bug fixes:

> - For SN/I scenarios, Id Web's `GetTokenAcquirer` now sets `SendX5C`. See issue [#2887](https://github.com/AzureAD/microsoft-identity-web/issues/2887) for details.

Reviewer: ```suggestion
- For SN/I scenarios, Id Web's `GetTokenAcquirer` now sets `SendX5C` in particular protocols. See issue [#2887](https://github.com/AzureAD/microsoft-identity-web/issues/2887) for details.
```

> - Fix for Instance/Tenant parsing for V2 authority. See PR [#2954](https://github.com/AzureAD/microsoft-identity-web/issues/2954) for details.

Reviewer: ```suggestion
- Fix for Instance/Tenant parsing for V2 authority (affected one Entra External IDs scenario). See PR [#2954](https://github.com/AzureAD/microsoft-identity-web/issues/2954) for details.
```

> - Fix a format exception: `The input string " was not in a correct format` due to incorrect regex. See issue [#2879](https://github.com/AzureAD/microsoft-identity-web/issues/2879) for details.

Reviewer: ```suggestion
- Fix regex that throwed a format exception: `The input string " was not in a correct format` when enabling *same-site cookie compatibility*. See issue [#2879](https://github.com/AzureAD/microsoft-identity-web/issues/2879) for details.
```

> - Id Web now has an upper bound set on its dependency on Microsoft.Identity.Abstractions to version 7x. See PR [#2962](https://github.com/AzureAD/microsoft-identity-web/pull/2962) for details.

Reviewer: ```suggestion
- Microsoft.Identity.Web 3.1.0 now has an upper bound set on its dependency on Microsoft.Identity.Abstractions to version 7x to avoid you Microsoft.Identity.Abstractions 8.0.0. See PR [#2962](https://github.com/AzureAD/microsoft-identity-web/pull/2962) for details.
```

  

### Fundamentals:
- Fix flakey tests: [#2972](https://github.com/AzureAD/microsoft-identity-web/pull/2972), [#2984](https://github.com/AzureAD/microsoft-identity-web/pull/2984), [#2982](https://github.com/AzureAD/microsoft-identity-web/issues/2982), 

> - Update to `AzureKeyVault@2`, [#2981](https://github.com/AzureAD/microsoft-identity-web/pull/2981).

Reviewer: ```suggestion
- Update to `AzureKeyVault@2` in AzureDevOps, [#2981](https://github.com/AzureAD/microsoft-identity-web/pull/2981).
```

- Update to .NET 9-preview7, [#2980](https://github.com/AzureAD/microsoft-identity-web/pull/2980) and [#2991](https://github.com/AzureAD/microsoft-identity-web/pull/2991).

> - Changes for a new release build: [#2974](https://github.com/AzureAD/microsoft-identity-web/pull/2974), [#2990](https://github.com/AzureAD/microsoft-identity-web/pull/2990)

Reviewer: ```suggestion
- It's now possible to build a specific version of Microsoft.Identity.Web based on specific versions of Microsoft.IdentityModel and Microsoft.Identity.Abstractions by specifying build variables on the dotnet pack command (MicrosoftIdentityModelVersion, MicrosoftIdentityAbstractionsVersions, and MicrosoftIdentityWebVersion): [#2974](https://github.com/AzureAD/microsoft-identity-web/pull/2974), [#2990](https://github.com/AzureAD/microsoft-identity-web/pull/2990)
```
